src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs

```diff
@@ -405,4 +411,28 @@ public void EnumerateLinks(Activity activity, int maxLinks)
             }
         }
     }
+
+    private struct ActivitySourceTagsEnumerationState
```

Comment:
This `struct` doesn't seem to be doing a whole lot. Did you do that because other code is? That might be carry-over from old enumeration optimizations. You could probably just make a static helper method for this.

Reply:
I felt the same way. As, other code implements it this way I followed the same format.